packages/server/utils/ServerHealthChecker.ts

```diff
@@ -0,0 +1,74 @@
+import Redis from 'ioredis'
+import sleep from '../../client/utils/sleep'
+import ServerAuthToken from '../database/types/ServerAuthToken'
+import {UserPresence} from '../graphql/private/mutations/connectSocket'
+import {disconnectQuery} from '../socketHandlers/handleDisconnect'
+import publishInternalGQL from './publishInternalGQL'
+import sendToSentry from './sendToSentry'
+
+const REDIS_URL = process.env.REDIS_URL!
+const SERVER_ID = process.env.SERVER_ID!
+
+// When a socket server starts up, it d
+export default class ServerHealthChecker {
+  publisher = new Redis(REDIS_URL, {connectionName: 'serverHealth_pub'})
+  subscriber = new Redis(REDIS_URL, {connectionName: 'serverHealth_sub'})
+  pendingPongs = new Set<string>()
+  constructor() {
+    this.subscriber.on('message', (_channel, message) => {
+      this.pendingPongs.delete(message)
+    })
+    this.subscriber.subscribe(`socketServerPong:${SERVER_ID}`)
+    this.publisher.sadd('socketServers', SERVER_ID)
+  }
+  async ping() {
+    const socketServers = await this.publisher.smembers('socketServers')
+    this.pendingPongs = new Set(...socketServers.filter((id) => id !== SERVER_ID))
+    this.publisher.publish('socketServerPing', SERVER_ID)
+    await sleep(500)
+    // if a server hasn't replied in 500ms, assume it is offline
+    const deadServerIds = [...this.pendingPongs]
+    await this.reportDeadServers(deadServerIds)
+    return deadServerIds
+  }
+
+  async reportDeadServers(deadServerIds: string[]) {
+    if (deadServerIds.length === 0) return
+    const authToken = new ServerAuthToken()
+    // remove the dead servers from the set of socket servers in redis
+    const writes = deadServerIds.map((id) => ['srem', 'socketServers', id])
+    await this.publisher.multi(writes).exec()
```

Comment:
+1 `await this.publisher.srem('socketServers', deadServerIds)` should work just fine here